connect/proxy/listener_test.go

```diff
@@ -223,4 +223,9 @@ func TestUpstreamListener(t *testing.T) {
 	// Check all the tx/rx counters got added
 	assertAllTimeCounterValue(t, sink, "consul.proxy.test.upstream.tx_bytes;src=web;dst_type=service;dst=db", 11)
 	assertAllTimeCounterValue(t, sink, "consul.proxy.test.upstream.rx_bytes;src=web;dst_type=service;dst=db", 11)
+
+	// Ensure that the listener is closed the port is actually released.
```

Comment:
```suggestion
	// Ensure that when the listener is closed the port is actually released.
```

(If I understand it correctly)